src/main/java/randoop/generation/GrtFuzzing.java

```diff
@@ -0,0 +1,580 @@
+package randoop.generation;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import randoop.main.GenInputsAbstract;
+import randoop.main.RandoopBug;
+import randoop.operation.CallableOperation;
+import randoop.operation.ConstructorCall;
+import randoop.operation.MethodCall;
+import randoop.operation.TypedClassOperation;
+import randoop.operation.TypedOperation;
+import randoop.sequence.Sequence;
+import randoop.sequence.Sequence.RelativeNegativeIndex;
+import randoop.sequence.Statement;
+import randoop.types.NonParameterizedType;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.Randomness;
+import randoop.util.SimpleArrayList;
+
+/**
+ * Implements the "GRT Impurity" component, as described in "GRT: Program-Analysis-Guided Random
+ * Testing" by Ma et. al (ASE 2015): <a
+ * href="https://people.kth.se/~artho/papers/lei-ase2015.pdf">...</a>.
+ *
+ * <p>The GRT Impurity component is a fuzzing mechanism that alters the states of (or creates new)
+ * input objects for methods under test to generate more object states and potentially trigger more
+ * branches, improving coverage for the program under test. Because it is a fuzzing mechanism, our
+ * implementation is called "GrtFuzzing".
+ *
+ * <p>[TODO: As described in the GRT paper, "GRT Impurity" also generates more effective tests with
```

Comment:
Do we want to implement this in the scope of our project?

Reply:
Actually, I don't even see in the GRT paper where pure methods are used. Just from the description, it looks like an impure method is always selected